src/app/qgisapp.cpp

```diff
@@ -1430,6 +1446,9 @@ QgisApp::QgisApp( QSplashScreen *splash, bool restorePlugins, bool skipVersionCh
     attributeTable( QgsAttributeTableFilterModel::ShowVisible );
   } );
 
+  QShortcut *shortcutTracing = new QShortcut( QKeySequence( tr( "Ctrl+Shift+." ) ), this );
```

Comment:
Mind adding "alt" in here too? This really isn't a shortcut you'd expect normal users to ever use, and we would want to minimise accidental triggers

Reply:
Hmm... I have tried to add Alt in as well - but hit an issue: for people who use multiple keyboard layouts (like me: English & Slovak), Alt+Shift is a commonly used keyboard shortcut to switch keyboard layouts (all the way back to windows 95). And this global shortcut gets priority over Ctrl+Alt+Shift+. defined here :-( Also, trying to press Ctrl + Alt + Shift at the same time, I can't find a way how to press them (at least on my keyboard) without curling my fingers in a very awkward way :)

So... can we live with Ctrl+Shift+. or do you have any other ideas how to make it harder to trigger accidentally? It feels that the keys are quite far away to be pressed by accident - and fortunately there isn't any destructive action going to happen (not even slowing down QGIS measurably).

Reply:
What about either:
1. hiding it behind a qsettings flag ("enable tracing")? 
2. make it show in the customisation dialog but without any initial shortcut assigned, so that it's an opt-in thing?

Reply:
Hiding it behind a qsettings flag would be probably a better solution...

One of the use cases is that I want to be able to instruct a random bug reporter with a problem with 3D to record a trace and attach it to the ticket. If it is possible to enable tracing in Options > Advanced then that should not be too difficult for bug reporters to do...

Reply:
I guess another option would be to put it in iface -- so you'd enable by pasting something in the python console like

   iface.startTrace()
   iface.saveTrace('c:/....')

Reply:
I took the approach with a QSettings flag that has to be enabled in Options > Advanced first...

The option with iface sounds also interesting for the future if we want to run this in a script...

Reply:
Great!

Reply:
Great!